cmake/3rdParty.cmake

```diff
@@ -122,7 +122,13 @@ function(ly_add_external_target)
             set(BASE_PATH "${LY_3RDPARTY_PATH}/${ly_add_external_target_3RDPARTY_DIRECTORY}")
 
         else()
-            ly_install_external_target(${ly_add_external_target_3RDPARTY_ROOT_DIRECTORY})
+            # only install external 3rdParty that are within the source tree
+            cmake_path(RELATIVE_PATH ly_add_external_target_3RDPARTY_ROOT_DIRECTORY 
+                BASE_DIRECTORY ${LY_ROOT_FOLDER}
+                OUTPUT_VARIABLE relative_path)
+            if(relative_path AND NOT relative_path MATCHES "^../")
+                ly_install_external_target(${ly_add_external_target_3RDPARTY_ROOT_DIRECTORY})
+            endif()
```

Comment:
@lumberyard-employee-dm this is probably the main thing I would like you to take a look at... 
So, the problem is that we have some 3rdParty that are part of the source tree (e.g. glad). And those need to be copied to the install layout. The ones coming from 3p package system dont need to be copied because they will be downloaded. The ones in the system (e.g. PIX, renderdoc, wwise, etc) dont need to be installed since they will be in the system (despite those that are build-time integrations cannot become part of the SDK unless they are enabled). 

Here I am relying on the path to the 3rdParty being inside the source tree to do that distinction. Sounds reasonable?

Reply:
Yes that sounds good.
We should use the [IS_PREFIX](https://cmake.org/cmake/help/git-stage/command/cmake_path.html#is-prefix) function though in to determine if the path is equal or within a base directory.